core/templates/services/attribution.service.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to handle the attribution experience.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { ContextService } from 'services/context.service';
+
+import { ExplorationSummaryBackendApiService } from 'domain/summary/exploration-summary-backend-api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AttributionService {
+  attributionModalIsShown: boolean = false;
+  authors: string[] = [];
+  explorationTitle: string = '';
+  constructor(
+    private contextService: ContextService,
+    private explorationSummaryBackendApiService: (
+      ExplorationSummaryBackendApiService)
+  ) {}
+
+  private _init(successCallback, errorCallback): void {
```

Comment:
Instead of passing sucessCallback and errorCallBack, can you just convert it to a promise?